### PR TITLE
Added Odd Even Linked List in C++

### LINKED ISSUE #4608: Odd Even Linked List in C++
## 💥 Proposal

Hi, I am a GSSOC'21 participant
I want to contribute the solution for the following problem statement in C++ language
"Given a singly linked list, rearrange it to group all the odd indices nodes together followed by the even indexed node"

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by sg-iitg**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added code for "odd even linked list" problem in C++ language
The program rearranges a given linked list so that all odd indexed nodes are arranged first, followed by the even indexed nodes in one iteration through the linked list

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Fixes #4608

### CODE REVIEW

**Comment by sg-iitg** [comment]
@ankitaggarwal23 @HarshCasper @siddharth25pandey 
Please review this pull request
Thankyou!

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -196,6 +196,7 @@
 - [In-order (LNR) Tree Traversal](graphs/inorder-traversal.cpp)
 - [Kosaraju Algorithm](graphs/Kosaraju_Algorithm.cpp)
 - [Kruskal's Algorithm](graphs/kruskal_Algorithm.cpp)
+- [Largest distance between 2 nodes of a tree](graphs/Largest_distance_between_2_nodes_in_a_tree.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Please revert this change. Its part of a separate PR

File: C-Plus-Plus/other/odd_even_linked_list.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        //pointer to next odd indexed element
+        co->next= co->next->next;
+        //pointer to next even indexed element
+        ce->next= ce->next->next;
+        co= co->next;
+        ce= ce->next;
+    }
+
+    //join the 2 linked list
+    co->next= head_even;
```
Comment (HarshCasper) [inline C-Plus-Plus/other/odd_even_linked_list.cpp]: Format the Code here

File: C-Plus-Plus/other/odd_even_linked_list.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    return head;
+}
+
+int main()
+{
+    int n;
```
Comment (HarshCasper) [inline C-Plus-Plus/other/odd_even_linked_list.cpp]: Can you use more meaningful variable name?

File: C-Plus-Plus/other/odd_even_linked_list.cpp [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    cin>>n;
+
+    int x;
```
Comment (HarshCasper) [inline C-Plus-Plus/other/odd_even_linked_list.cpp]: Can you use more meaningful variable name?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**